### PR TITLE
Fixed using of host and ssh_host flags

### PR DESCRIPTION
**Pull request by toha10**
This fix tries to fix a mess with ssh_host param.
Now it gets default value from host flag.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:27: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'
************* Module cloudferrylib.os.actions.convert_file
cloudferrylib/os/actions/convert_file.py:15: [W0612(unused-variable), ConvertFile.run] Unused variable 'instance'
************* Module cloudferrylib.os.actions.merge_base_and_diff
cloudferrylib/os/actions/merge_base_and_diff.py:11: [W0612(unused-variable), MergeBaseDiff.run] Unused variable 'instance'
************* Module cloudferrylib.os.actions.transport_ephemeral
cloudferrylib/os/actions/transport_ephemeral.py:180: [W0612(unused-variable), TransportEphemeral.copy_ephemeral_iscsi_to_ceph] Unused variable 'inst_id'
cloudferrylib/os/actions/transport_ephemeral.py:195: [W0612(unused-variable), TransportEphemeral.rebase_diff] Unused variable 'instance_id'
************* Module cloudferrylib.os.actions.utils
cloudferrylib/os/actions/utils.py:38: [E1102(not-callable), transfer_file_to_file] utils.up_ssh_tunnel is not callable
************* Module cloudferrylib.os.actions.upload_file_to_image
cloudferrylib/os/actions/upload_file_to_image.py:37: [W0612(unused-variable), UploadFileToImage.run] Unused variable 'instance'
************* Module cloudferrylib.os.network.nova_network
cloudferrylib/os/network/nova_network.py:58: [W0221(arguments-differ), NovaNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/nova_network.py:61: [W0221(arguments-differ), NovaNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
************* Module cloudferrylib.utils.drivers.ssh_ceph_to_ceph
cloudferrylib/utils/drivers/ssh_ceph_to_ceph.py:30: [W0221(arguments-differ), SSHCephToCeph.transfer] Arguments number differs from overridden 'transfer' method
cloudferrylib/utils/drivers/ssh_ceph_to_ceph.py:35: [E1129(not-context-manager), SSHCephToCeph.transfer] Context manager 'tuple' doesn't implement __enter__ and __exit__.
************* Module cloudferrylib.utils.drivers.ssh_ceph_to_file
cloudferrylib/utils/drivers/ssh_ceph_to_file.py:32: [E1102(not-callable), SSHCephToFile.transfer] utils.up_ssh_tunnel is not callable
************* Module cloudferrylib.utils.drivers.ssh_file_to_ceph
cloudferrylib/utils/drivers/ssh_file_to_ceph.py:36: [E1129(not-context-manager), SSHFileToCeph.transfer] Context manager 'tuple' doesn't implement __enter__ and __exit__.
************* Module cloudferrylib.utils.drivers.ssh_file_to_file
cloudferrylib/utils/drivers/ssh_file_to_file.py:36: [E1102(not-callable), SSHFileToFile.transfer] utils.up_ssh_tunnel is not callable
cloudferrylib/utils/drivers/ssh_file_to_file.py:80: [E1129(not-context-manager), SSHFileToFile.transfer_direct] Context manager 'tuple' doesn't implement __enter__ and __exit__.
************* Module cloudferrylib.utils.ssh_util
cloudferrylib/utils/ssh_util.py:18: [W0403(relative-import), ] Relative import 'cmd_cfg', should be 'cloudferrylib.utils.cmd_cfg'
cloudferrylib/utils/ssh_util.py:19: [W0403(relative-import), ] Relative import 'utils', should be 'cloudferrylib.utils.utils'
cloudferrylib/utils/ssh_util.py:23: [W0613(unused-argument), SshUtil.__init__] Unused argument 'host'
************* Module cloudferrylib.os.storage.cinder_storage
cloudferrylib/os/storage/cinder_storage.py:139: [W0221(arguments-differ), CinderStorage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/storage/cinder_storage.py:318: [W0613(unused-argument), CinderStorage.convert_snapshot] Unused argument 'cloud'
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:412: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:51: [W0212(protected-access), GlanceImageTestCase.setUp] Access to a protected member _resp of a client class
************* Module tests.cloudferrylib.os.storage.test_cinder_storage
tests/cloudferrylib/os/storage/test_cinder_storage.py:241: [W0703(broad-except), CinderStorageTestCase.test_get_volume_path_iscsi_error] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/actions/pre_transport_instance.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
+# Copyright (c) 2014 Mirantis Inc.
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/pre_transport_instance.py]: This file seem to be unrelated to the purpose of patch

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:27: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'
************* Module cloudferrylib.os.actions.convert_file
cloudferrylib/os/actions/convert_file.py:15: [W0612(unused-variable), ConvertFile.run] Unused variable 'instance'
************* Module cloudferrylib.os.actions.merge_base_and_diff
cloudferrylib/os/actions/merge_base_and_diff.py:11: [W0612(unused-variable), MergeBaseDiff.run] Unused variable 'instance'
************* Module cloudferrylib.os.actions.upload_file_to_image
cloudferrylib/os/actions/upload_file_to_image.py:37: [W0612(unused-variable), UploadFileToImage.run] Unused variable 'instance'
************* Module cloudferrylib.os.actions.utils
cloudferrylib/os/actions/utils.py:38: [E1102(not-callable), transfer_file_to_file] utils.up_ssh_tunnel is not callable
************* Module cloudferrylib.os.actions.transport_ephemeral
cloudferrylib/os/actions/transport_ephemeral.py:180: [W0612(unused-variable), TransportEphemeral.copy_ephemeral_iscsi_to_ceph] Unused variable 'inst_id'
cloudferrylib/os/actions/transport_ephemeral.py:195: [W0612(unused-variable), TransportEphemeral.rebase_diff] Unused variable 'instance_id'
************* Module cloudferrylib.os.network.nova_network
cloudferrylib/os/network/nova_network.py:58: [W0221(arguments-differ), NovaNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/nova_network.py:61: [W0221(arguments-differ), NovaNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
************* Module cloudferrylib.utils.drivers.ssh_ceph_to_ceph
cloudferrylib/utils/drivers/ssh_ceph_to_ceph.py:30: [W0221(arguments-differ), SSHCephToCeph.transfer] Arguments number differs from overridden 'transfer' method
cloudferrylib/utils/drivers/ssh_ceph_to_ceph.py:35: [E1129(not-context-manager), SSHCephToCeph.transfer] Context manager 'tuple' doesn't implement __enter__ and __exit__.
************* Module cloudferrylib.utils.drivers.ssh_ceph_to_file
cloudferrylib/utils/drivers/ssh_ceph_to_file.py:32: [E1102(not-callable), SSHCephToFile.transfer] utils.up_ssh_tunnel is not callable
************* Module cloudferrylib.utils.drivers.ssh_file_to_ceph
cloudferrylib/utils/drivers/ssh_file_to_ceph.py:36: [E1129(not-context-manager), SSHFileToCeph.transfer] Context manager 'tuple' doesn't implement __enter__ and __exit__.
************* Module cloudferrylib.utils.drivers.ssh_file_to_file
cloudferrylib/utils/drivers/ssh_file_to_file.py:36: [E1102(not-callable), SSHFileToFile.transfer] utils.up_ssh_tunnel is not callable
cloudferrylib/utils/drivers/ssh_file_to_file.py:80: [E1129(not-context-manager), SSHFileToFile.transfer_direct] Context manager 'tuple' doesn't implement __enter__ and __exit__.
************* Module cloudferrylib.utils.ssh_util
cloudferrylib/utils/ssh_util.py:18: [W0403(relative-import), ] Relative import 'cmd_cfg', should be 'cloudferrylib.utils.cmd_cfg'
cloudferrylib/utils/ssh_util.py:19: [W0403(relative-import), ] Relative import 'utils', should be 'cloudferrylib.utils.utils'
cloudferrylib/utils/ssh_util.py:23: [W0613(unused-argument), SshUtil.__init__] Unused argument 'host'
************* Module cloudferrylib.os.storage.cinder_storage
cloudferrylib/os/storage/cinder_storage.py:140: [W0221(arguments-differ), CinderStorage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/storage/cinder_storage.py:319: [W0613(unused-argument), CinderStorage.convert_snapshot] Unused argument 'cloud'
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:420: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:51: [W0212(protected-access), GlanceImageTestCase.setUp] Access to a protected member _resp of a client class
************* Module tests.cloudferrylib.os.storage.test_cinder_storage
tests/cloudferrylib/os/storage/test_cinder_storage.py:243: [W0703(broad-except), CinderStorageTestCase.test_get_volume_path_iscsi_error] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
@toha10 please check grizzly to juno job, it failed with:
```
VM with name keypair_test_server and ip: 192.168.1.108 is not accessible via key pair
-------------------- >> begin captured stdout << ---------------------
[192.168.1.108] run: pwd
```

**Comment by toha10** [comment]
@roman-verchikov I think that the fail in grizzly to juno migration job is not related to this patch.The patch influence only on tasks related to interactions with openstack nodes (ssh_host), but here i assume FT tries to check VM's availability via floating ip using ssh.

**Comment by toha10** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:27: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'
************* Module cloudferrylib.os.actions.convert_file
cloudferrylib/os/actions/convert_file.py:15: [W0612(unused-variable), ConvertFile.run] Unused variable 'instance'
************* Module cloudferrylib.os.actions.merge_base_and_diff
cloudferrylib/os/actions/merge_base_and_diff.py:11: [W0612(unused-variable), MergeBaseDiff.run] Unused variable 'instance'
************* Module cloudferrylib.os.actions.upload_file_to_image
cloudferrylib/os/actions/upload_file_to_image.py:37: [W0612(unused-variable), UploadFileToImage.run] Unused variable 'instance'
************* Module cloudferrylib.os.actions.utils
cloudferrylib/os/actions/utils.py:38: [E1102(not-callable), transfer_file_to_file] utils.up_ssh_tunnel is not callable
************* Module cloudferrylib.os.actions.transport_ephemeral
cloudferrylib/os/actions/transport_ephemeral.py:180: [W0612(unused-variable), TransportEphemeral.copy_ephemeral_iscsi_to_ceph] Unused variable 'inst_id'
cloudferrylib/os/actions/transport_ephemeral.py:195: [W0612(unused-variable), TransportEphemeral.rebase_diff] Unused variable 'instance_id'
************* Module cloudferrylib.os.network.nova_network
cloudferrylib/os/network/nova_network.py:58: [W0221(arguments-differ), NovaNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/nova_network.py:61: [W0221(arguments-differ), NovaNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
************* Module cloudferrylib.utils.drivers.ssh_ceph_to_ceph
cloudferrylib/utils/drivers/ssh_ceph_to_ceph.py:30: [W0221(arguments-differ), SSHCephToCeph.transfer] Arguments number differs from overridden 'transfer' method
cloudferrylib/utils/drivers/ssh_ceph_to_ceph.py:35: [E1129(not-context-manager), SSHCephToCeph.transfer] Context manager 'tuple' doesn't implement __enter__ and __exit__.
************* Module cloudferrylib.utils.drivers.ssh_ceph_to_file
cloudferrylib/utils/drivers/ssh_ceph_to_file.py:32: [E1102(not-callable), SSHCephToFile.transfer] utils.up_ssh_tunnel is not callable
************* Module cloudferrylib.utils.drivers.ssh_file_to_ceph
cloudferrylib/utils/drivers/ssh_file_to_ceph.py:36: [E1129(not-context-manager), SSHFileToCeph.transfer] Context manager 'tuple' doesn't implement __enter__ and __exit__.
************* Module cloudferrylib.utils.drivers.ssh_file_to_file
cloudferrylib/utils/drivers/ssh_file_to_file.py:36: [E1102(not-callable), SSHFileToFile.transfer] utils.up_ssh_tunnel is not callable
cloudferrylib/utils/drivers/ssh_file_to_file.py:80: [E1129(not-context-manager), SSHFileToFile.transfer_direct] Context manager 'tuple' doesn't implement __enter__ and __exit__.
************* Module cloudferrylib.utils.ssh_util
cloudferrylib/utils/ssh_util.py:18: [W0403(relative-import), ] Relative import 'cmd_cfg', should be 'cloudferrylib.utils.cmd_cfg'
cloudferrylib/utils/ssh_util.py:19: [W0403(relative-import), ] Relative import 'utils', should be 'cloudferrylib.utils.utils'
cloudferrylib/utils/ssh_util.py:23: [W0613(unused-argument), SshUtil.__init__] Unused argument 'host'
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:420: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
************* Module cloudferrylib.os.storage.cinder_storage
cloudferrylib/os/storage/cinder_storage.py:140: [W0221(arguments-differ), CinderStorage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/storage/cinder_storage.py:319: [W0613(unused-argument), CinderStorage.convert_snapshot] Unused argument 'cloud'
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:51: [W0212(protected-access), GlanceImageTestCase.setUp] Access to a protected member _resp of a client class
************* Module tests.cloudferrylib.os.storage.test_cinder_storage
tests/cloudferrylib/os/storage/test_cinder_storage.py:243: [W0703(broad-except), CinderStorageTestCase.test_get_volume_path_iscsi_error] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1, can be merged
